app/services/publisher_statement_getter.rb

```diff
@@ -7,6 +7,23 @@ class Statement
 
     UPHOLD_CONTRIBUTION = "uphold_contribution".freeze
     UPHOLD_CONTRIBUTION_SETTLEMENT = "uphold_contribution_settlement".freeze
+
+    def fee?
+      transaction_type == 'fees'
+    end
+
+    def eyeshade?
+      transaction_type == 'contribution_settlement' || transaction_type == 'referral_settlement' || fee?
+    end
+
+    def earning_period
+      # If the transaction_type is from Eyeshade this means the period was for the previous month
+      if eyeshade?
+        (created_at - 1.month).at_beginning_of_month
```

Comment:
Would you double check to prove this works all the time? Is 1.month automagically the previous month? Would `1.month.ago` provide a different answer?

Reply:
For future reference - 
1.month is automagically the correct month before, [here's the implementation](https://github.com/rails/rails/blob/9ae511f55047b2380224ad993a7d0ed97dbcf944/activesupport/lib/active_support/core_ext/integer/time.rb#L20), from what I can recall it basically subtracts 28-29ish days from the date? 

Doing `.ago` converts it from `Date.today` and looks to [be implemented here](https://api.rubyonrails.org/v5.1/classes/Time.html#method-i-ago)